### PR TITLE
make text selectable/copyable and replace links with hyperlinks

### PR DESCRIPTION
**Pull request by jaycenhorton**
This change replaces plain-text links in changelogs to be hyperlinks and also enables copying/selecting of text in changelogs.

Note: I created a test release [here](https://github.com/jaycenhorton/obsidian-plugin-update-tracker/releases/tag/rehype-links2) and tested via BRAT

## Before (additionally, text is un-selectable)

image

## After (additionally, text is selectable for copying)

image

### CODE REVIEW

**Comment by swar8080** [comment]
thanks for the improvements! i'll try taking a closer look later this week

**Comment by jaycenhorton** [comment]
@swar8080 FYI the remark plugin was definitely just a nice to have for me, the single line of css I changed would also be completely sufficient as it enables selecting and copying of the text so I can copy paste the url i the browser. If it makes it easier to review I can open that as a separate PR.

**Comment by swar8080** [comment]
@jaycenhorton I think a separate PR for just the text selection for now would be great. I can create a new release once that's done

The rehype change would be nice but i'd want to set-up a quick performance test and check for other edge cases with URLs. My guess is that raw URLs are rare since obsidian users would likely use markdown or the github URL icon when writing release notes

**Comment by claremacrae** [comment]
tl;dr: 

- it works great
- and I feel it would be a great help for reviewing updates to plugins that use GitHub's 'Generate release notes' facility

### Testing the PR

I tested this out by building the code and copying it to my main vault.

It works great for raw URLs that are not in list items (which were already shown as links).

I checked:

- The visual appearance
- That the new links do actually work

Before:

- image

After:

- image

It presumably won't linkify the strings that GitHub auto-links in release pages `@username` and `#1234` - but at least now but those will be reachable  by first following the link to the PR...

### Value of the change

@swar8080  I do appreciate that in an ideal world, it's good to have PRs each change only one thing.

But this is such a great quality-of-life improvement for users that maybe you could consider merging it anyway?
It can be reverted easily if there is a real-world problem.

### Thoughts on swar8080's comments

> The rehype change would be nice but i'd want to set-up a quick performance test and check for other edge cases with URLs.

Regarding performance, on my 4-year Mac the rendering was near-instantaneous: even if it did slow things down slightly, I suspect most users would be grateful for the more useful behaviour.

> My guess is that raw URLs are rare since obsidian users would likely use markdown or the github URL icon when writing release notes

The GitHub 'Generate release notes' feature generates a load of text like this:

```
* feat: Add date picker in Reading mode and Tasks Query results by @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3038
* feat: Add a date picker to the Edit Task modal by @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3052
* fix: Improve Edit Task modal calendar 'button' layout by @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3058
* fix: Improve Edit Task modal calendar button behaviour by @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3059
* vault: Add date picker smoke tests & fix GitHub-created sample vault by @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3064
* fix: remove datepicker pending usability fixes by @ilandikov and @claremacrae in https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3067
```

Which GitHub renders in releases like this (when logged in as @claremacrae):

image

And the current release of this plugin renders like this:

image


This PR means that the PR numbers, at least, can be navigated to, which makes it way easier to decide whether to install updates for those plugins that use this format in their Release Notes.

**Comment by claremacrae** [comment]
> And the current release of this plugin renders like this:

And after this PR, this plugin renders as:

image

**Comment by claremacrae** [comment]
> It works great for raw URLs that are not in list items (which were already shown as links).

I revisited that, to try and figure out why some URLs were previously hyperlinked...

It turns out that the ones in the screenshot that were linked before this PR were in `<...>`

```
**Full Changelog** (including changes too small to note above, and that do not affect the behaviour of the plugin itself):
- <https://github.com/obsidian-tasks-group/obsidian-tasks/compare/7.10.2...7.11.0>

**🙏 Support Tasks development**
- <https://github.com/sponsors/claremacrae>
```

See https://github.com/obsidian-tasks-group/obsidian-tasks/releases/edit/7.11.0

**Comment by claremacrae** [comment]
It looks like this was merged in the following, possibly to avoid merge conflicts?

- #102

And released in:

- [V1.6.0](https://github.com/swar8080/obsidian-plugin-update-tracker/releases/tag/1.6.0)

Thank you for the credit in the release notes! ❤️

**Comment by jaycenhorton** [comment]
Apologies for not getting around to it before you beat me to it, but thanks @swar8080 for including this feature. Closing this as redundant.
